pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1334,15 +1334,26 @@ func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta
 //  topology of all the "terms" for the given "pod".
 func (c *PodAffinityChecker) nodeMatchesAllTopologyTerms(pod *v1.Pod, topologyPairs *topologyPairsMaps, nodeInfo *schedulercache.NodeInfo, terms []v1.PodAffinityTerm) bool {
 	node := nodeInfo.Node()
-	for _, term := range terms {
-		if topologyValue, ok := node.Labels[term.TopologyKey]; ok {
-			pair := topologyPair{key: term.TopologyKey, value: topologyValue}
-			if _, ok := topologyPairs.topologyPairToPods[pair]; !ok {
-				return false
-			}
-		} else {
+OUTERLOOP:
+	for idx, term := range terms {
+		topologyValue, ok := node.Labels[term.TopologyKey]
+		if !ok {
 			return false
 		}
+
+		pair := topologyPair{key: term.TopologyKey, value: topologyValue}
+		podSet, ok := topologyPairs.topologyPairToPods[pair]
+		if !ok {
+			return false
+		}
+		// "idx" should be found in "podSet[<any pod>]"
+		for _, bitwisedIdx := range podSet {
+			// this indicates current "idx" is satisfied
+			if bitwisedIdx|(1<<uint(idx)) == bitwisedIdx {
+				continue OUTERLOOP
```

Comment:
Why don't you just `break` here? Those labels remind me of `goto`.

Reply:
I see. I would add flags to cover that. Please don't use labels.

Reply:
sure, I actually used https://github.com/kubernetes/kubernetes/pull/68725/commits/20b71555fba4520df7c0e9c2b2f6714c8f4678e5 to "try out" the `continue LABEL` directive. Before that, I used a flag to achieve same thing.